### PR TITLE
:gear: More preparation for Remix v2 (1)

### PR DESCRIPTION
**Pull request by agoose77**
More work on preparing for v2 in Remix, fixing `CatchBoundary` and some other small changes.

### CODE REVIEW

**Comment by agoose77** [comment]
I noticed during upgrade work that despite `v2_routeConvention` being set, I had to backport to v1 routing to fix some navigation bugs. I won't tackle that here, as I don't firmly understand the problem yet.

**Comment by agoose77** [comment]
@stevejpurves can we merge this? It silences warnings and doesn't break anything to my knowledge.

File: themes/article/remix.config.prod.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ module.exports = {
   appDirectory: 'app',
   assetsBuildDirectory: 'public/build',
   serverBuildPath: 'build/index.js',
+  serverModuleFormat: 'cjs',
```
Comment (rowanc1) [inline themes/article/remix.config.prod.js:6]:
```suggestion
```

**Review by stevejpurves — APPROVED** [review]
tested out locally - looks good!

**Pull request merged**